Add receptorctl status documentation (#997)
ansible/receptor: docs/source/receptorctl/receptorctl_status.rst

--------------------------
Receptorctl status command

> --------------------------

Reviewer: I'd probably change this to just `status` or maybe `status` command.

With this "Receptorctl x command" format it will get difficult to parse through the doc at a high level because all the sections will look the same.

```
Receptorctl status command
Receptorctl foo command
Receptorctl bar command
```

The preceding format results in the command getting lost in the surrounding text, which means the reader needs to work a bit harder. From the title, we've got the context that we're talking about Receptorctl client commands. So a simpler title for each section would make more sense to me.

```
status
foo
bar
```

.. contents::
   :local:

> ``receptorctl status`` shows the Receptor network status

Reviewer: ```suggestion
``receptorctl status`` displays the status of the Receptor network.
```

> The command syntax is: ``receptorctl --socket=<socket_path> status [--json]``

Reviewer: ```suggestion
Command syntax: ``receptorctl --socket=<socket_path> status [--json]``
```

> ``socket_path`` is the control socket address to connect to Receptor.

Reviewer: ```suggestion
``socket_path`` is the control socket address for the Receptor connection.
```

Author: Also updated source

>    It defaults to Unix socket.

Reviewer: ```suggestion
   The default is ``unix:`` for a Unix socket.
```

Author: also updated in source

>    Use tcp:// for TCP socket.

Reviewer: ```suggestion
   Use ``tcp://`` for a TCP socket.
```

Author: updated source also

>    The corresponding environment variable is ``RECEPTORCTL_SOCKET``

Reviewer: ```suggestion
   The corresponding environment variable is ``RECEPTORCTL_SOCKET``.
```

``ss --listening --processes --unix 'src = unix:<socket_path>'``
Netid         State          Recv-Q         Send-Q                   Local Address:Port                     Peer Address:Port        Process

> u_str         LISTEN         0              4096                   /tmp/local.sock 38130170                            * 0            users:(("receptor",pid=3226769,fd=7))

Reviewer: Should these three lines go into a code-block?

Author: absolutely, great suggestion!

``ps -fp $(pidof receptor)``
``lsof -p <<pid>``

> ``--json`` will format the output as JSON.

Reviewer: ```suggestion
The ``--json`` option returns the output in JSON format.
```

>    The default is human-readable.

Reviewer: ```suggestion
   The default output format is human-readable.
```

>    Using this argument allows output to be machine consumable.  For example, piping into ``jq``

Reviewer: ```suggestion
   Using this argument allows output to be machine consumable.  For example, piping into ``jq``.
```

> The output is divided into sections listed below.  Field values may be listed in their own section.

Reviewer: ```suggestion
The output is divided into sections listed below.
Field values may be listed in their own section.
```

Columns are the actual JSON node values.

^^^^^^^^^^^^

> Node Section

Reviewer: ```suggestion
Node output
```

^^^^^^^^^^^^

> .. list-table:: Node Section

Reviewer: ```suggestion
.. list-table:: Node output
```

      :header-rows: 1
      :widths: auto

      * - Column
        - Description
      * - ``."NodeID"``

>         - Node Identifier.

Reviewer: ```suggestion
        - Node identifier.
```

>         - Node Identifier

Reviewer: ```suggestion
        - Unique identifier for the node.
```

      * - ``."SystemCPUCount"``

>         - Number of CPUs on node

Reviewer: ```suggestion
        - Number of CPUs on the node.
```

Reviewer: Is this the number of CPUs allocated/available or actually in use?

      * - ``.SystemMemoryMiB"``

>         - Available Mmemory (MiB) of the node.

Reviewer: ```suggestion
        - Available memory (MiB) of the node.
```

>         - Memory (MiB)

Reviewer: ```suggestion
        - Available memory (MiB).
```

Reviewer: I'm guessing it's available memory, not something like total memory used.

Author: Correct - https://github.com/pbnjay/memory/tree/master

      * - ``."Version"``

>         - Receptor version

Reviewer: ```suggestion
        - Receptor version.
```

^^^^^^^^^^^^^^^^^^^

> Connections Section

Reviewer: ```suggestion
Connections output
```

Reviewer: Global comment for this one. "Section" isn't very descriptive imo. The heading level should denote the fact it is a section of the document. It seems like "output" is more relevant.

^^^^^^^^^^^^^^^^^^^

.. list-table:: Connections section
    :header-rows: 1
    :widths: auto

    * - Column
      - Description
    * - ``."Connections"``

>       - Connections.

Reviewer: Is this number of active connections or something more like configured number of allowed connections?

Author: list of active connections

>       - Connections section

Reviewer: Hmm. Maybe I'm mistaken about s/section/output?

Author: back to section

    * - ``."Connections"."Cost"``

>       - Cost to reach NodeID

Reviewer: Is there another word here? What do we mean by "cost"? Is it latency in terms of milliseconds?

In general better to not use the same word as the thing you're trying to describe.

>       - Metric (route preference) to reach NodeID.

Reviewer: Can we provide an example here? Are there several possible metrics?

Author: This refers to the connection weight.  We will defer this for now

    * - ``."Connections"."NodeID"``

>       - Node ID.

Reviewer: Is this any different to the NodeID in the Node section?

Author: Deferred this discussion

^^^^^^^^^^^^^^^^^^^^^^^^^

> Known Connections section

Reviewer: ```suggestion
Known connections section
```

^^^^^^^^^^^^^^^^^^^^^^^^^

> .. list-table:: Known Connections section

Reviewer: ```suggestion
.. list-table:: Known connections section
```
